zeppelin-zengine/src/main/java/org/apache/zeppelin/remoteworks/RemoteWorksManager.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.remoteworks;
+
+import org.apache.zeppelin.interpreter.InterpreterContextRunner;
+import org.apache.zeppelin.interpreter.RemoteWorksController;
+import org.apache.zeppelin.notebook.Note;
+import org.apache.zeppelin.notebook.Notebook;
+import org.apache.zeppelin.notebook.Paragraph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Zeppelin Server RemoteWorkController Singleton.
+ */
+public class RemoteWorksManager {
+  private static final Logger LOG = LoggerFactory.getLogger(RemoteWorksManager.class);
+  private static NotebookJobManager instance;
+
+  public RemoteWorksManager(Notebook notebook) {
+    if (RemoteWorksManager.instance == null) {
+      RemoteWorksManager.instance = new NotebookJobManager(notebook);
+    }
+  }
+
+  public static NotebookJobManager getInstance() {
+    return RemoteWorksManager.instance;
+  }
+
+  private class NotebookJobManager implements RemoteWorksController {
```

Comment:
What is purpose of having `NotebookJobManager` private class and `RemoteWorksController` interface?
I think all the methods in the `NotebookJobManager` can be moved to `RemoteWorksManager`, and `RemoteWorksController` interface can be removed. 

I think that's simple and easier to understand.